addon/ng2/utilities/ast-utils.ts

```diff
@@ -0,0 +1,55 @@
+import * as ts from 'typescript';
+import { InsertChange } from './change';
+
+/**
+* Find all nodes from the AST in the subtree of node of SyntaxKind kind.
+* @param node 
+* @param kind
+* @return all nodes of kind kind, or [] if none is found
+*/
+export function findNodes(node: ts.Node, kind: ts.SyntaxKind): ts.Node[] {
+  if (!node) {
+    return [];
+  }
+  let arr = [];
+  if (node.kind === kind) {
+    arr.push(node);
+  }
+  node.getChildren().forEach(child => arr.push.apply(arr, findNodes(child, kind)));
```

Comment:
Might be a bit more complex (reduce operations are harder to grasp at first), but this line could be simplified by:

```typescript
return node.getChildren().reduce((prev, current) => prev.concat(findNodes(current)), []);
```

Here's the documentation on reduce: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/Reduce. It's quite powerful.